### ISSUE TITLE
error in marytts cart DecisionNode (classic Greek word in German voice)

### ISSUE THREAD

**Issue by aitorme** [marytts/marytts#428]
while running the convert text to phoneme task on de/bleeding, I got the following error:
```
<stacktrace>marytts.exceptions.SynthesisException: cannot process
	at marytts.LocalMaryInterface.process(LocalMaryInterface.java:440)
	at marytts.LocalMaryInterface.generateXML(LocalMaryInterface.java:319)
	at marytts.MaryInterface$generateXML.call(Unknown Source)
	at Text2Phonemes$_main_closure1.doCall(Text2Phonemes.groovy:31)
	at sun.reflect.GeneratedMethodAccessor14.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:292)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1015)
	at groovy.lang.Closure.call(Closure.java:423)
	at groovy.lang.Closure.call(Closure.java:439)
	at org.codehaus.groovy.runtime.ResourceGroovyMethods.eachFile(ResourceGroovyMethods.java:1038)
	at org.codehaus.groovy.runtime.ResourceGroovyMethods.eachFile(ResourceGroovyMethods.java:1056)
	at org.codehaus.groovy.runtime.dgm$790.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:271)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:53)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
	at Text2Phonemes.main(Text2Phonemes.groovy:27)
Caused by: java.lang.Exception: Module JPhonemiser_de: Problem processing the data.
	at marytts.server.Request.processOneChunk(Request.java:550)
	at marytts.server.Request.processOrLookupOneChunk(Request.java:385)
	at marytts.server.Request.process(Request.java:327)
	at marytts.LocalMaryInterface.process(LocalMaryInterface.java:438)
	... 21 more
Caused by: java.lang.ClassCastException: marytts.cart.DecisionNode$ByteDecisionNode cannot be cast to marytts.cart.LeafNode$StringAndFloatLeafNode
	at marytts.modules.phonemiser.TrainedLTS.predictPronunciation(TrainedLTS.java:145)
	at marytts.language.de.JPhonemiser.phonemise(JPhonemiser.java:353)
	at marytts.language.de.JPhonemiser.process(JPhonemiser.java:233)
	at marytts.server.Request.processOneChunk(Request.java:548)
	... 24 more
</stacktrace>
```

**Comment by seblemaguer** [comment]
can you be more explicit : which text (as it's a wikipedia article can you put the dump on pastebin or gist or something like that and add the url ?) ...

**Comment by aitorme** [comment]
the article is [Astat.txt](https://github.com/marytts/marytts/files/24328/Astat.txt),  the error is thrown in the very first line, since there is the original Ancient Greek word in the file. This is the error output when processing it into phonemes:
```
<html><body><h1>Internal server error</h1><p>Processing failed.<pre>java.lang.AssertionError
	at marytts.cart.CART.interpretToNode(CART.java:137)
	at marytts.modules.phonemiser.TrainedLTS.predictPronunciation(TrainedLTS.java:156)
	at marytts.language.de.JPhonemiser.phonemise(JPhonemiser.java:355)
	at marytts.language.de.JPhonemiser.process(JPhonemiser.java:233)
	at marytts.server.Request.processOneChunk(Request.java:566)
	at marytts.server.Request.processOrLookupOneChunk(Request.java:403)
	at marytts.server.Request.process(Request.java:337)
	at marytts.server.http.SynthesisRequestHandler.process(SynthesisRequestHandler.java:261)
	at marytts.server.http.SynthesisRequestHandler.handleClientRequest(SynthesisRequestHandler.java:91)
	at marytts.server.http.BaseHttpRequestHandler.handle(BaseHttpRequestHandler.java:138)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler$RequestHandlerAdaptor.handle(BufferingHttpServiceHandler.java:189)
	at org.apache.http.nio.protocol.SimpleNHttpRequestHandler.handle(SimpleNHttpRequestHandler.java:51)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.processRequest(AsyncNHttpServiceHandler.java:453)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.inputReady(AsyncNHttpServiceHandler.java:316)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler.inputReady(BufferingHttpServiceHandler.java:131)
	at org.apache.http.impl.nio.DefaultNHttpServerConnection.consumeInput(DefaultNHttpServerConnection.java:173)
	at org.apache.http.impl.nio.DefaultServerIOEventDispatch.inputReady(DefaultServerIOEventDispatch.java:147)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:161)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:335)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:275)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:542)
	at java.lang.Thread.run(Thread.java:745)
</pre></body></html>
```
There is another word in Greek alphabet in the text, which throws the exact same error if it is the first word of the input text, but it is simply skipped when there is another word before it given as the input.

**Comment by seblemaguer** [comment]
fixed in commit 09285bb by ignoring the word
